### ISSUE TITLE
Replace the Ollama HTTP embedding call with local ONNX inference

### ISSUE THREAD

**Issue by shadowbrok3r** [shadowbrok3r/MastertechProject#241]
## What

`database/src/schema/utilities.rs:25` posts to a remote Ollama endpoint on every embedded write:

```rust
pub const OLLAMA_EMBEDDINGS_URL: &str = "https://ollama.shadowbroker.app/api/embeddings";
```

Run `nomic-embed-text` locally as ONNX instead — in Rust, with `ort` + `tokenizers`.

## Why

- It is a network round-trip **inside a write path**. `diagnostic_entry`, `stress_test_run` and `hardware_component` all embed at write time.
- It fails often enough that `backfill_missing_embeddings()` exists to clean up after it, and `spawn_embedding_backfill()` has to be kicked off to catch the rows it missed.
- Rows written while the endpoint is down carry no vector, so they are invisible to HNSW search until a backfill runs.
- It is a hard dependency on an external host for a core write.

## Why not `ml::` in SurrealQL

Sentence-transformer ONNX takes `input_ids` / `attention_mask`. SurrealML's `buffered_compute` value map is a flat dict of named floats with stored normalizers — the tokenization step does not fit. `raw_compute` from Rust does, which is why this belongs in the app and not in the database.

Also relevant: the `ml::` function exists in SurrealDB 3.x, but the SurrealML packaging library is still preview/unstable, and we are on `3.3.0-beta.2`.

## Scope

- [ ] `nomic-embed-text` exported to ONNX, 768-dim output verified against the current endpoint
- [ ] Tokenizer + inference behind the existing `embed_text()` signature so call sites do not change
- [ ] Output parity checked — cosine similarity against existing stored vectors on a sample, since re-embedding with a mismatched model silently degrades every HNSW index
- [ ] Decide whether existing vectors need regenerating
- [ ] Remote endpoint kept as a fallback, or removed once parity holds
- [ ] `backfill_missing_embeddings` retired if local inference cannot fail the same way

## Notes

Affects three HNSW indexes, all 768-dim COSINE F32: `diag_embedding`, `hw_component_embedding`, `strun_embedding`.
